evalml/tests/automl_tests/test_automl.py

```diff
@@ -1063,6 +1079,88 @@ class Pipeline2(DummyPipeline):
 
 
+@pytest.mark.parametrize("problem_type", ["binary", "multiclass", "regression"])
+def test_percent_better_than_baseline_computed_for_all_objectives(problem_type,
+                                                                  dummy_binary_pipeline_class,
+                                                                  dummy_multiclass_pipeline_class,
+                                                                  dummy_regression_pipeline_class,
+                                                                  X_y_binary):
+
+    X, y = X_y_binary
+
+    problem_type_enum = handle_problem_types(problem_type)
+
+    pipeline_class = {"binary": dummy_binary_pipeline_class,
+                      "multiclass": dummy_multiclass_pipeline_class,
+                      "regression": dummy_regression_pipeline_class}[problem_type]
+    baseline_pipeline_class = {"binary": "evalml.pipelines.ModeBaselineBinaryPipeline",
+                               "multiclass": "evalml.pipelines.ModeBaselineMulticlassPipeline",
+                               "regression": "evalml.pipelines.MeanBaselineRegressionPipeline",
+                               }[problem_type]
+
+    class DummyPipeline(pipeline_class):
+        name = "Dummy 1"
+        problem_type = problem_type_enum
+
+        def fit(self, *args, **kwargs):
+            """Mocking fit"""
+
+    core_objectives = get_core_objectives(problem_type)
+    mock_scores = {obj.name: i for i, obj in enumerate(core_objectives)}
+    mock_baseline_scores = {obj.name: i + 1 for i, obj in enumerate(core_objectives)}
+    answer = {obj.name: obj.calculate_percent_difference(mock_scores[obj.name],
+                                                         mock_baseline_scores[obj.name]) for obj in core_objectives}
+
+    mock_score_1 = MagicMock(return_value=mock_scores)
+    DummyPipeline.score = mock_score_1
+
+    automl = AutoMLSearch(problem_type=problem_type, max_iterations=2,
+                          allowed_pipelines=[DummyPipeline], objective="auto")
+
+    with patch(baseline_pipeline_class + ".score", return_value=mock_baseline_scores):
+        automl.search(X, y, data_checks=None)
+        assert len(automl.results['pipeline_results']) == 2, "This tests assumes only one non-baseline pipeline was run!"
+        pipeline_results = automl.results['pipeline_results'][1]
+        assert pipeline_results["percent_better_than_baseline_all_objectives"] == answer
+        assert pipeline_results['percent_better_than_baseline'] == pipeline_results["percent_better_than_baseline_all_objectives"][automl.objective.name]
+
+
+@pytest.mark.parametrize("fold_scores", [[2, 4, 6], [np.nan, 4, 6]])
+@patch("evalml.pipelines.ModeBaselineBinaryPipeline.score", return_value={'Log Loss Binary': 1, 'F1': 1})
+@patch('evalml.pipelines.BinaryClassificationPipeline.score')
+@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
+def test_percent_better_than_baseline_scores_different_folds(mock_fit,
```

Comment:
I noticed that all our other unit tests assume the same score is returned on each fold so I decided to add one where the score is different for each fold.

Reply:
Great!